### PR TITLE
SHOULD fix sepchems

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request

separated chems works again, i think.

## Why It's Good For The Game

Makes sepchems work. If i did this right, instead of a separated chems plant falling to the ground and being deleted immediately after squash, it should instead work like a grenade- detonating in 3-6 seconds if primed or if squashed.

## Changelog
:cl:
fix: fixed sep chems so it actually does something again
balance: but keeps it nerfed to grenade levels
/:cl:

### CODE REVIEW

**Comment by zeskorion** [comment]
Ugh... gonna throw in the towel and go to bed for tonight. Ill fix my shitcode tommorow

**Comment by zeskorion** [comment]
Wait the last minute change before i went to bed worked


Kek

File: code/modules/hydroponics/grown.dm [hunk 1]
```diff
@@ -89,6 +89,12 @@
 /obj/item/reagent_containers/food/snacks/grown/attack_self(mob/user)
 	if(seed && seed.get_gene(/datum/plant_gene/trait/squash))
 		squash(user)
+	if(seed.get_gene(/datum/plant_gene/trait/noreact))
+		user.visible_message("<span class='warning'>[user] shakes [src] vigorously!</span>", "<span class='userdanger'>You shake [src] vigorously!</span>")
+		log_bomber(user, "primed a", src, "for detonation")
```
Comment (qwertyquerty) [inline code/modules/hydroponics/grown.dm]: Sep chems not always a bomb

File: code/modules/hydroponics/grown.dm [hunk 2]
```diff
@@ -89,6 +89,12 @@
 /obj/item/reagent_containers/food/snacks/grown/attack_self(mob/user)
 	if(seed && seed.get_gene(/datum/plant_gene/trait/squash))
 		squash(user)
+	if(seed.get_gene(/datum/plant_gene/trait/noreact))
+		user.visible_message("<span class='warning'>[user] shakes [src] vigorously!</span>", "<span class='userdanger'>You shake [src] vigorously!</span>")
+		log_bomber(user, "primed a", src, "for detonation")
+		if(iscarbon(user))
+			var/mob/living/carbon/C = user
+			C.throw_mode_on()
```
Comment (qwertyquerty) [inline code/modules/hydroponics/grown.dm]: Don't turn throw mode on, again sep chems doesn't always have to be treated like a grenade

File: code/modules/hydroponics/grown.dm (hunk 1 above)
Comment (zeskorion) [reply code/modules/hydroponics/grown.dm]: done

File: code/modules/hydroponics/grown.dm (hunk 2 above)
Comment (zeskorion) [reply code/modules/hydroponics/grown.dm]: well, works exactly like a chem grenade, but also done

File: code/modules/hydroponics/grown.dm [hunk 3]
```diff
@@ -117,11 +119,18 @@
...
-	for(var/A in T)
-		reagents.reaction(A)
-
+	if(!seed.get_gene(/datum/plant_gene/trait/noreact))
+		reagents.reaction(T)
+		for(var/A in T)
+			reagents.reaction(A)
+		qdel(src)
+	if(seed.get_gene(/datum/plant_gene/trait/noreact))
+		playsound(loc, 'sound/effects/fuse.ogg', 75, 1, -3)
```
Comment (qwertyquerty) [inline code/modules/hydroponics/grown.dm:128]: No lol. It's a plant dude.

File: code/modules/hydroponics/grown.dm [hunk 4]
```diff
@@ -117,11 +119,18 @@
...
+		addtimer(CALLBACK(src, .proc/prime), rand(30, 60))
```
Comment (qwertyquerty) [inline code/modules/hydroponics/grown.dm:129]: Given how I've heard these plants aren't easy to pull off, I would suggest changing the range to 20, 50

File: code/modules/hydroponics/plant_genes.dm [hunk 5]
```diff
@@ -315,11 +318,10 @@
 /datum/plant_gene/trait/noreact/on_new(obj/item/reagent_containers/food/snacks/grown/G, newloc)
 	..()
 	ENABLE_BITFIELD(G.reagents.flags, NO_REACT)
-
-/datum/plant_gene/trait/noreact/on_squash(obj/item/reagent_containers/food/snacks/grown/G, atom/target)
+	
+/datum/plant_gene/trait/noreact/on_prime(obj/item/reagent_containers/food/snacks/grown/G, atom/target)
```
Comment (qwertyquerty) [inline code/modules/hydroponics/plant_genes.dm:322]: Why did you change the name of this?

File: code/modules/hydroponics/grown/misc.dm [hunk 6]
```diff
@@ -141,18 +141,18 @@
...
-		prime()
+		cprime()
 	if(!QDELETED(src))
 		qdel(src)
 
 /obj/item/reagent_containers/food/snacks/grown/cherry_bomb/ex_act(severity)
 	qdel(src) //Ensuring that it's deleted by its own explosion. Also prevents mass chain reaction with piles of cherry bombs
 
-/obj/item/reagent_containers/food/snacks/grown/cherry_bomb/proc/prime()
+/obj/item/reagent_containers/food/snacks/grown/cherry_bomb/proc/cprime()
```
Comment (qwertyquerty) [inline code/modules/hydroponics/grown/misc.dm:155]: cprime? What?

File: code/modules/hydroponics/grown/citrus.dm [hunk 7]
```diff
@@ -135,7 +135,7 @@
 /obj/item/reagent_containers/food/snacks/grown/firelemon/ex_act(severity)
 	qdel(src) //Ensuring that it's deleted by its own explosion
 
-/obj/item/reagent_containers/food/snacks/grown/firelemon/proc/prime()
+/obj/item/reagent_containers/food/snacks/grown/firelemon/proc/lprime()
```
Comment (qwertyquerty) [inline code/modules/hydroponics/grown/citrus.dm:138]: lprime????

File: code/modules/hydroponics/grown.dm [hunk 8]
```diff
@@ -89,6 +89,8 @@
 /obj/item/reagent_containers/food/snacks/grown/attack_self(mob/user)
 	if(seed && seed.get_gene(/datum/plant_gene/trait/squash))
 		squash(user)
+	if(seed.get_gene(/datum/plant_gene/trait/noreact))
+		user.visible_message("<span class='warning'>[user] shakes [src] vigorously!</span>", "<span class='userdanger'>You shake [src] vigorously!</span>")
```
Comment (qwertyquerty) [inline code/modules/hydroponics/grown.dm:93]: Why?

File: code/modules/hydroponics/grown.dm (hunk 4 above)
Comment (zeskorion) [reply code/modules/hydroponics/grown.dm:129]: sepchems aint too hard to get, it's the high death mixes that are (and are still rushable in a half hour or so by a competent botanist)

a 2 second fusetime is a liability to the user, however, at least if you use it like a grenade

File: code/modules/hydroponics/plant_genes.dm (hunk 5 above)
Comment (zeskorion) [reply code/modules/hydroponics/plant_genes.dm:322]: because I added a new on_prime proc. Squashing deletes the plant, and timers don't work unless they call a proc- the reason sepchems broke is because the plant was deleted before it mixed.

File: code/modules/hydroponics/grown/misc.dm (hunk 6 above)
Comment (zeskorion) [reply code/modules/hydroponics/grown/misc.dm:155]: conflicts

File: code/modules/hydroponics/grown.dm (hunk 8 above)
Comment (zeskorion) [reply code/modules/hydroponics/grown.dm:93]: this makes it so you can use it in-hand, instead of just throwing it at a mob to activate it- this is why the timer was so long

File: code/modules/hydroponics/grown.dm (hunk 3 above)
Comment (zeskorion) [reply code/modules/hydroponics/grown.dm:128]: it needs some form of sharp auditory feedback when it's primed. Fuse is more organic than the grenade beeps, which combustible lemon uses for some reason, but still designed for this purpose. Reagent bubbling and the backpack rustle were two more sounds i thought of that could be used, though.

File: code/modules/hydroponics/grown/citrus.dm (hunk 7 above)
Comment (zeskorion) [reply code/modules/hydroponics/grown/citrus.dm:138]: conflicts

File: code/modules/hydroponics/grown.dm (hunk 3 above)
Comment (qwertyquerty) [reply code/modules/hydroponics/grown.dm:128]: You realize that it's just a plant getting squashed on the ground right.

Comment (zeskorion) [reply code/modules/hydroponics/grown.dm:128]: see below on using in hand. a bit of auditory feedback is good regardless, though

**Comment by qwertyquerty** [comment]
I'm not a fan of this implementation. I think it should be activated via squashing. I will not merge this simply due to low code quality. Please just fix the existing implementation.

**Comment by zeskorion** [comment]
> I'm not a fan of this implementation. I think it should be activated via squashing. I will not merge this simply due to low code quality. Please just fix the existing implementation.

but it *is* activated via squashing, just simply made analogous to grenades, as it should be.

**Comment by zeskorion** [comment]
regardless, ill make a new pr for this without grenade functionality

**Comment by qwertyquerty** [comment]
Activated by thrown squashing

**Comment by zeskorion** [comment]
> Activated by thrown squashing

these ARE activated by thrown squashing still though, it's just not as viable. it's changing how they were used because the original use was cancer and unbalanced in the first place, but making them function like mass produced chem grenades removes that
